crates/uv-fs/src/lib.rs

```diff
@@ -282,18 +282,26 @@ pub struct LockedFile(fs_err::File);
 impl LockedFile {
     pub fn acquire(path: impl AsRef<Path>, resource: impl Display) -> Result<Self, std::io::Error> {
         let file = fs_err::File::create(path.as_ref())?;
+        debug!("Trying to lock if free: {}", path.as_ref().user_display());
         match file.file().try_lock_exclusive() {
             Ok(()) => Ok(Self(file)),
-            Err(err) if err.kind() == std::io::ErrorKind::WouldBlock => {
+            Err(err) => {
+                // Log error code and enum kind to help debugging more exotic failures
+                debug!("Try lock error, waiting for exclusive lock: {:?}", err);
                 warn_user!(
                     "Waiting to acquire lock for {} (lockfile: {})",
                     resource,
                     path.user_display(),
                 );
-                file.file().lock_exclusive()?;
+                file.file().lock_exclusive().map_err(|err| {
+                    // Not an fs_err method, we need to build our own path context
```

Comment:
@konstin can you open an upstream issue to add?

Reply:
I don't think that's in scope for either project